src/tms_service.ts

```diff
@@ -110,6 +147,97 @@ export class TMSService extends AbstractEmsService {
     this._config = config;
   }
 
+  /*
+  This method returns an array of objects with the layers and the new
+  layout['text-field'] to apply using map.setLayoutProperty
+  */
+  public static transformLanguageProperty(
+    layer: LayerSpecification,
+    lang: string
+  ): DataDrivenPropertyValueSpecification<FormattedSpecification> | undefined {
+    const supportedLang = this.SupportedLanguages.find((l) => l.key === lang);
+    if (supportedLang === undefined) {
```

Comment:
Throwing an error may be too strong here. Or maybe I just have to add a catch in Kibana if `i18n.locale` is not a supported language. What do you think?

Reply:
OK so the function will return `undefined` for non supported languages and it will be up to the method user how to handle upwards :+1: